source/package-manager/index.js

```diff
@@ -0,0 +1,65 @@
+import fs from 'node:fs';
+import path from 'node:path';
+import semver from 'semver';
+import {npmConfig, yarnBerryConfig, pnpmConfig, yarnConfig, npmConfigNoLockfile} from './configs.js';
+
+/**
+@param {string} rootDir
+@param {import('./types.d.ts').PackageManagerConfig} config
+*/
+export function findLockFile(rootDir, config) {
+	return config.lockfiles
+		.map(filename => ({filename, filepath: path.resolve(rootDir || '.', filename)}))
+		.find(({filepath}) => fs.existsSync(filepath));
```

Comment:
Since `filename` isn't being used:

```suggestion
		.map(lockfile => path.resolve(rootDir || '.', filename))
		.find(filepath => fs.existsSync(filepath));
```